assets/js/components/ModulesList.js

```diff
@@ -0,0 +1,174 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import {
+	showErrorNotification,
+} from '../util';
+import Data from 'googlesitekit-data';
+import Link from './Link';
+import ModuleIcon from './ModuleIcon';
+import GenericError from './notifications/generic-error';
+import ModuleSettingsWarning from './notifications/module-settings-warning';
+import VisuallyHidden from './VisuallyHidden';
+import { STORE_NAME as CORE_MODULES } from '../googlesitekit/modules/datastore/constants';
+
+const { useSelect, useDispatch } = Data;
+
+function ModulesList( { moduleSlugs } ) {
+	const { activateModule } = useDispatch( CORE_MODULES );
+	const modulesData = useSelect( ( select ) => select( CORE_MODULES ).getModules() ?? {} );
+
+	/**
+	 * Handle setup module click event.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param {string} slug Module slug.
+	 */
+	const handleSetupModule = useCallback( async ( slug ) => {
+		try {
+			const { response } = await activateModule( slug );
+
+			// Redirect to ReAuthentication URL
+			global.location.assign( response.moduleReauthURL );
+		} catch ( err ) {
+			showErrorNotification( GenericError, {
+				id: 'setup-module-error',
+				title: __( 'Internal Server Error', 'google-site-kit' ),
+				description: err.message,
+				format: 'small',
+				type: 'win-error',
+			} );
+		}
+	}, [ activateModule ] );
+
+	// Filter specific modules
+	const moduleObjects = Array.isArray( moduleSlugs ) && moduleSlugs.length
+		? moduleSlugs
+			.filter( ( slug ) => modulesData.hasOwnProperty( slug ) )
+			.reduce( ( acc, slug ) => ( { ...acc, [ slug ]: modulesData[ slug ] } ), {} )
+		: modulesData;
+
+	// Filter out internal modules.
+	const modules = Object.values( moduleObjects ).filter( ( module ) => ! module.internal );
+
+	// Map of slug => name for every module that is active and completely set up.
+	const completedModuleNames = modules
+		.filter( ( module ) => module.active && module.setupComplete )
+		.reduce( ( completed, module ) => {
+			completed[ module.slug ] = module.name;
+			return completed;
+		}, {} );
+
+	// Sort modules and exclude those that required other modules to be set up.
+	// Logic still in place below in case we want to add blocked modules back.
+	const sortedModules = modules
+		.filter( ( module ) => 0 === module.dependencies.length )
+		.sort( ( module1, module2 ) => module1.sort - module2.sort );
+
+	return (
+		<div className="googlesitekit-modules-list">
+			{ map( sortedModules, ( module ) => {
+				let blockedByParentModule = false;
+				let parentBlockerName = '';
+				const {
+					slug,
+					name,
+					active: isConnected,
```

Comment:
`active` means that the module is active but not connected (a user hasn't configured it yet). We need to use the `connected` property to determine if the module connected or not.